crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -1650,6 +1650,11 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                 ruff::rules::assignment_in_assert(checker, expr);
             }
         }
+        Expr::Lambda(lambda_expr) => {
+            if checker.enabled(Rule::ReimplementedOperator) {
+                refurb::rules::reimplemented_operator(checker, &lambda_expr.into());
+            }
+        }
```

Comment:
This needed to be moved from `deferred_lambdas.rs` to `expressions.rs` because otherwise the semantic model doesn't understand that the lambda it's analysing is ever inside a class scope. As far as I can tell, there's no particular advantage associated with deferring the analysis of this rule to after the semantic model has been fully built; it never analyses any bindings or uses of bindings

Reply:
Thanks for writing this comment. It saved me quiet some time guessing why it was moved.

Reply:
it took _me_ a little while to figure out why the semantic model didn't think lambdas inside class scopes were inside class scopes 😅